packages/cubejs-server-core/src/core/OrchestratorApi.ts

```diff
@@ -230,6 +231,18 @@ export class OrchestratorApi {
     }
   }
 
+  public async fetchSchema(dataSource: string, securityContext?: { [key: string]: any; }) {
+    const queryCache = this.orchestrator.getQueryCache();
+
+    const queryKey = this.orchestrator.getQueryCache()
+      .getKey('FETCH_SCHEMA', getCacheHash({ dataSource, securityContext }));
+
+    const queue = await queryCache.getQueue(dataSource);
+    const tablesSchema = await queue.executeQueryInQueue('query', queryKey, { tablesSchema: true });
+
+    return tablesSchema;
```

Comment:
It would be great if you will move this logic to the `QueryCache` class and create `QueryOrchestrator.fetchSchema` method that will call it. `OrchestrationApi` is more like `facade`, and we prefare to keep it as simple as it possible.

Reply:
It also allows you to keep internal `getCacheHash` inside of its module.